File: Lab4/Python/func.py

def inputT():
    time = float(input("Enter t in seconds: "))
    while True:
        if time > 0:
            return time
        else:
            time = float(input("Wrong input. Try again: "))

def inputN():
    count = int(input("Enter the number of points: "))
    while True:
        if count > 0:
            return count
        else:
            count = float(input("Wrong input. Try again: "))


def outputCoordinates(array):
    for i in range(len(array)):
        array[i].InitialCoordinates()
        array[i].FinalCoordinates()
        print()



def MinD(array):
    minD = array[0].GetDistance(array[1].GetX(), array[1].GetY(), array[1].GetZ())
    for i in range(len(array)-1):
        for j in range(i+1,len(array)):
            temp = array[i].GetDistance(array[j].GetX(), array[j].GetY(), array[j].GetZ())
            if temp < minD:
                minD = temp
    return minD

File: Lab5/func.cpp
#include "func.h"
PointXYZ* XYZ_Array(int size, int time)
{
	PointXYZ* Array = new PointXYZ[size];
	for (int i = 0; i < size; i++)
	{
		PointXYZ point;
		point.coordinates(time);
		Array[i] = point;
	}
	return Array;
}
PointXY* XY_Array(int size, int time)
{
	PointXY* Array = new PointXY[size];
	for (int i = 0; i < size; i++)
	{
		PointXY point;
		point.coordinates(time);
		Array[i] = point;
	}
	return Array;
}
int input()
{
	int num = 0;
	do
	{
		cout << "Enter the number of points: "; cin >> num;
	} while (num <= 0);
	return num;
}
int inputT()
{
	cout << "Enter t in seconds: ";
	int time;
	cin >> time;
	while (true)
	{
		if (time > 0)return time;
		else cin >> time;
	}
}
void outputPoint(Point* Array, int size, int time)
{
	for (int i = 0; i < size; i++)
	{
		cout << "M" << i + 1 << ":" << endl;
		cout << "Initial coordinates ofthe point: " << Array[i].GetStart() << endl;
		cout << "Coordinates of the point in " << time << " seconds: " << Array[i].GetFinal() << endl;
		cout << endl;
	}
}
double MinDistance(Point* Array, int size)
{
	double minD = Array[0].GetDistance(Array[1].GetX(), Array[1].GetY(), Array[1].GetZ());
	for (int i = 0; i < size-1; i++)
	{
		for (int j = i+1; j < size; j++)
		{
			double temp = Array[i].GetDistance(Array[j].GetX(), Array[j].GetY(), Array[j].GetZ());
			if (temp < minD)minD = temp;
		}
	}
	return minD;
}
